static/app/components/draggableTabs/draggableTabList.tsx

```diff
@@ -24,6 +38,202 @@ interface BaseDraggableTabListProps extends DraggableTabListProps {
   items: DraggableTabListItemProps[];
 }
 
+function useIsDragging(ref: React.RefObject<HTMLElement>) {
+  const [isDragging, setIsDragging] = useState(false);
+
+  useEffect(() => {
+    const element = ref.current;
+
+    if (!element) {
+      return () => {};
+    }
+
+    const handlePointerDown = () => {
+      setIsDragging(true);
+    };
+    const handlePointerUp = () => {
+      setIsDragging(false);
+    };
+    element.addEventListener('pointerdown', handlePointerDown);
+    document.addEventListener('pointerup', handlePointerUp);
+
+    return () => {
+      element.removeEventListener('pointerdown', handlePointerDown);
+      document.removeEventListener('pointerup', handlePointerUp);
+    };
+  }, [ref]);
+
+  return isDragging;
+}
+
+function useOverflowingTabs({state}: {state: TabListState<DraggableTabListItemProps>}) {
```

Comment:
This hook determines which tabs are overflowing. We can't use `IntersectionObserver` like the current Tabs component, because we will need to move the tabs around based on which one is selected. It's also different because the "Add View" button is to the right of the tabs, so we can't used a static container for the IntersectionObserver.